File: rules/non-determinism/python/synchronization-with-sleep.py
from selenium import webdriver
# rule-id: synchronization-with-sleep
webdriver.set_page_load_timeout(30)
webdriver.set_page_load_timeout(0.3)
# rule-id: synchronization-with-sleep
webdriver.set_script_timeout(20)
webdriver.set_script_timeout(0.2)

# rule-id: synchronization-with-sleep
await asyncio.sleep(1)

import time
# rule-id: synchronization-with-sleep
time.sleep(10)
# rule-id: synchronization-with-sleep
time.sleep(0.2)

def a():
    return 10

# OK
t.sleep(a())

# OK
t.sleep(some_var)

File: rules/non-determinism/java/synchronization-with-sleep.java
// rule-id: synchronization-with-sleep
Thread.sleep(10*1000);

public void Test()
{
    var sut = new SomeService();

    sut.FirePeriodicJob();

    // Give the job time to complete
    // rule-id: synchronization-with-sleep
    Thread.Sleep(10000);

    /* Verify the effects of the job */
}

public void Test()
{
    var sut = new SomeService();

    sut.FirePeriodicJob();

    // Give the job time to complete
    // rule-id: synchronization-with-sleep
    Task.Delay(10000).Wait();

    /* Verify the effects of the job */
}
